velox/type/Type.cpp

```diff
@@ -690,6 +690,7 @@ VELOX_DEFINE_SCALAR_ACCESSOR(BOOLEAN);
 VELOX_DEFINE_SCALAR_ACCESSOR(TINYINT);
 VELOX_DEFINE_SCALAR_ACCESSOR(SMALLINT);
 VELOX_DEFINE_SCALAR_ACCESSOR(BIGINT);
+VELOX_DEFINE_SCALAR_ACCESSOR(HUGEINT);
```

Comment:
I think we can't do this because HUGEINT creates a DecimalType, not a ScalarType specialization.

Cc: @majetideepak has more context.

Reply:
This should be okay. Won't impact DecimalType.
But HUGEINT TypeKind was added only to back the LongDecimal logical type.
There is not much support for HUGEINT as a first-class scalar type. For example, none of the vector functions have signatures for HUGEINT. Some scalar functions due to templating might work out of the box.